cwms-data-api/src/test/java/cwms/cda/api/VersionedTimeseriesControllerTestIT.java

```diff
@@ -0,0 +1,664 @@
+package cwms.cda.api;
+
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.http.HttpServletResponse;
+
+import cwms.cda.api.enums.VersionType;
+import cwms.cda.formatters.Formats;
+import fixtures.TestAccounts;
+
+import io.javalin.core.validation.JavalinValidation;
+import io.restassured.filter.log.LogDetail;
+import org.apache.commons.io.IOUtils;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+@Tag("integration")
+public class VersionedTimeseriesControllerTestIT extends DataApiTestIT {
+
+    private static final String OFFICE = "SWT";
+    private static final String locationId = "TsVersionedTestLoc";
+    private static final String tsId = locationId + ".Flow.Inst.1Hour.0.raw";
+    private static final String BEGIN_STR = "2008-05-01T15:00:00Z";
+    private static final String END_STR = "2008-05-01T17:00:00Z";
+
+    private Map<String, String> createParamMap() {
+        return new HashMap<String, String>()
+        {
+            {
+                put("office", OFFICE);
+                put("name", tsId);
+                put("begin", BEGIN_STR);
+                put("end", END_STR);
+                put("version-date", "2021-06-21T08:00:00-0000[UTC]");
+            }
+        };
+    }
+
+    private Map<String, String> replaceEndParam(Map<String, String> paramMap, String endDate) {
```

Comment:
You shouldn't have to "remove" the old key.  Calling put with the key and newValue will overwrite the old value.  
The method is a little weird - it boils down to just one line so its not shorter to call this than to just put that line of code.  There are other places in the test where you directly modify paramMap so I'd suggest removing this method and just inlining the paramMap.put("end", endDate) call

Reply:
done